### PR TITLE
Update RTree.java

### PR DESCRIPTION
**Pull request by paul150792**
if the "execute" method from the procedure returns "false", the searching will stop

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
   <groupId>net.sf.jsi</groupId>
   <artifactId>jsi</artifactId>
-  <version>1.1.0-SNAPSHOT</version>
+  <version>1.1.0-IRIAN</version>
```
Comment (aled) [inline pom.xml]: What is this?

File: pom.xml [hunk 2]
```diff
@@ -206,20 +208,6 @@
...
-          <executions>
-              <execution>
-                  <id>sign-artifacts</id>
-                  <phase>verify</phase>
-                  <goals>
-                      <goal>sign</goal>
-                  </goals>
-              </execution>
-          </executions>
-      </plugin>
```
Comment (aled) [inline pom.xml]: Why is this removed?

File: pom.xml [hunk 3]
```diff
@@ -272,42 +260,10 @@
...
-    <!-- ssh -t aled,jsi@shell.sourceforge.net create -->
-    <site>
-      <id>jsi.sourceforge.net</id>
-      <url>scp://shell.sourceforge.net/home/project-web/jsi/htdocs</url>
-    </site>
-
+      <repository>
+          <id>irian-3rd-party</id>
+          <url>http://maven.irian.at/content/repositories/thirdparty/</url>
+      </repository>
```
Comment (aled) [inline pom.xml]: ?

**Comment by aled** [comment]
Hi, thanks for making the effort to submit a pull request, but there are a couple of problems:
1) Whitespace in code is changed. Please make sure you don't change any lines of code (including whitespace) unless it is part of your patch.
2) Changes to pom.xml - I'm not sure what these are, can you explain what you have changed and why?

**Comment by paul150792** [comment]
Hi,

I've changed the pom.xml (locally) for two reasons:
1) 
```java
-      <plugin>
-          <groupId>org.apache.maven.plugins</groupId>
-          <artifactId>maven-gpg-plugin</artifactId>
-          <version>1.5</version>
-          <executions>
-              <execution>
-                  <id>sign-artifacts</id>
-                  <phase>verify</phase>
-                  <goals>
-                      <goal>sign</goal>
-                  </goals>
-              </execution>
-          </executions>
-      </plugin>
```
I deleted this part because it threw exceptions when compiling it (mvn clean install)

2) I uploaded the jar file in my own maven.server, but sadly I forgot to revert the pom.xlm before the push. Sorry!


I did only a small change to the code, which is part of my patch. 
1) I changed a little the method
```java
nearestN(Point p, TIntProcedure v, int n, float distance) 
```
so that it will stop on 
```java
Proceduce {
       execute(int i) {
              return false; 
        }
}
```

2)
I implemented the method 
```java
nearestNInKMDistance(Point p, TIntProcedure v, int count, float furthestDistanceKM)
```
the only difference between this method and the other is the fact that distances are in km.
this method uses following methods 
```java
createNearestNDistanceQueueInKM(Point p, int count, PriorityQueue distanceQueue, float furthestDistanceKM)


getDistanceFromLatLonInKm(float lat1, float lon1, float lat2, float lon2)
```

**Comment by aled** [comment]
Had a quck look, I think it looks fine in general. I'll merge it if you can redo the pull request without the whitespace changes and without the pom.xml changes. Not too keen on the code duplication but that can be refactored later.

Where did you get the algorithm for calculating the distance? Only took a quick look but it looks like it assumes a spherical earth? Maybe that could be reflected in the method name as it may be nice to have a WGS-84 version at some point.

I also thought the logical endpoint of this (in the future) would be making a subclass of RTree that uses the great circle distance calculation throughout the class, and does validation on the input coordinates to make sure they are actual lat/longs, and correctly handles rectangles that cross 0 degrees longitude.

**Comment by paul150792** [comment]
I found the algorithm [here] (http://stackoverflow.com/questions/27928/how-do-i-calculate-distance-between-two-latitude-longitude-points). Yes, it's assuming that the earth is spherical.

90% of my implemented methods are duplication, because I needed them very fast and that's why I duplicated them, but of course a re-factorization will solve this problem.

I will make tomorrow another pull request (today I don't have the necessary time for it)

I would like to help you with the implementation of new functionality. Are you interested?

**Comment by paul150792** [comment]
I managed to do now the changes, but on maven clean install I receive following error:

```java
Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:1.5:sign (sign-artifacts) on project jsi: Exit code: 2 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

but it works if I delete the plugin:
```java
org.apache.maven.plugins:maven-gpg-plugin
```

**Comment by aled** [comment]
The pull request is still showing 1423 added lines and 1290 deleted lines (It looks like you have reformatted the source to have indents of 4 spaces instead of 2)

**Comment by paul150792** [comment]
I think that it happened because I arranged the code with using Intellij. (Alt + i)

**Comment by paul150792** [comment]
I changed the whitespaces from 4 to 2. now it should be ok.

**Comment by aled** [comment]
Nope, still *hundreds* of whitespace changes there. Honestly if you have let your IDE reformat the code you need to either manually revert all the unwanted changes, or make a new branch and copy just your changes in.

**Comment by aled** [comment]
BTW, you could comment out that section of the pom.xml that causes a build error if you like.
